src/components/controller/AppController.js

```diff
@@ -1,49 +1,162 @@
 import React from 'react';
 import RightSidebar from '../views/rightsidebar/RightSidebar';
-import Choropleth from "../views/visualization/Choropleth";
-import GraphComponent from "../views/visualization/GraphComponent";
-import { expenditure_data } from "../../data/expenditure_data";
+import ReportView from "../views/reportview/ReportView";
+import { appController } from "../ConfigMap";
 
-const exp_data = expenditure_data;
 class AppController extends React.Component {
     constructor(props) {
         super(props);
         this.state = { 
-            budgetAttr:"BE",
+            heirarchyLevel : this.props.params.heirarchy_level,
+            categorySlug : this.props.params.category,
+            recordSlug : this.props.params.record,
+            series : this.props.route.panelName,
             viewBy: "choropleth",
+            budgetAttr : "BE",
```

Comment:
This is part of the bigger issue we are facing. To what extent can we make our React Components flexible is a subjective matter, which depends on the release cycle. While rigid components are less generic, they are easier to build. While a flexible component is more extensible it requires more time, which would affect the release cycle.